src/usePreviousDistinct/usePreviousDistinct.ts

```diff
@@ -0,0 +1,31 @@
+import { useRef } from 'react';
+import { useFirstMountState } from '..';
+
+export type Predicate<T> = (prev: T | undefined, next: T) => boolean;
+
+const checkIfStrictlyEqual = <T>(prev: T | undefined, next: T) => prev === next;
```

Comment:
I'd move it to `src/uril/const` and named it `isStrictEqual` and more broad arguments types, as it can be used in other hooks in future (or even we should check if anywhere else such prediacte is already used)
```ts
const isStrictEqual = (a: any, b: any): boolean => a === b;
```